src/test/java/bio/terra/service/resourcemanagement/flight/AzureStorageMonitoringStepProviderTest.java

```diff
@@ -0,0 +1,56 @@
+package bio.terra.service.resourcemanagement.flight;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+import bio.terra.service.resourcemanagement.azure.AzureMonitoringService;
+import bio.terra.stairway.Step;
+import java.util.List;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+@Tag("bio.terra.common.category.Unit")
+class AzureStorageMonitoringStepProviderTest {
+
+  private static final List<Class<? extends Step>> EXPECTED_STANDARD_STEPS =
+      List.of(CreateLogAnalyticsWorkspaceStep.class, CreateDiagnosticSettingStep.class);
+
+  private static final List<Class<? extends Step>> EXPECTED_PROTECTED_DATA_STEPS =
+      List.of(
+          CreateLogAnalyticsWorkspaceStep.class,
+          CreateDiagnosticSettingStep.class,
+          CreateExportRuleStep.class,
+          CreateSentinelStep.class,
+          CreateSentinelAlertRulesStep.class,
+          CreateSentinelNotificationRuleStep.class);
+
+  @Mock AzureMonitoringService monitoringService;
+
+  AzureStorageMonitoringStepProvider stepProvider;
+
+  @BeforeEach
+  void setUp() {
+    stepProvider = new AzureStorageMonitoringStepProvider(monitoringService);
+  }
+
+  @Test
+  void testConfigureStandardSteps() {
+    assertThat(
+        "only standard steps returned",
+        stepProvider.configureSteps(false).stream().map(s -> s.step().getClass()).toList(),
+        is(EXPECTED_STANDARD_STEPS));
+  }
+
+  @Test
+  void testConfigureProtectedDataSteps() {
+    assertThat(
+        "protected data steps also returned",
+        stepProvider.configureSteps(true).stream().map(s -> s.step().getClass()).toList(),
+        is(EXPECTED_PROTECTED_DATA_STEPS));
+  }
```

Comment:
If you change `configureSteps()` as I suggested above, here's how to test them. You can either use a mock with `verify`, or collect the results and compare them as you were doing before.
```suggestion
  @Test
  void testConfigureStandardSteps() {
    interface AddStep extends BiConsumer<Step, RetryRule> {};
    AddStep addStep = mock(AddStep.class);
    stepProvider.configureSteps(false, addStep);
    verify(addStep).accept(any(CreateLogAnalyticsWorkspaceStep.class), any());
    verify(addStep).accept(any(CreateDiagnosticSettingStep.class), any());
  }

  @Test
  void testConfigureProtectedDataSteps() {
    List<Class<? extends Step>> steps = new ArrayList<>();
    stepProvider.configureSteps(true, (step, retryRule) -> steps.add(step.getClass()));
    assertThat("protected data steps also returned", steps, is(EXPECTED_PROTECTED_DATA_STEPS));
  }
```

Reply:
Oh I mean, I get how to do it, I just think the way it is now is a little simpler.  I guess it's just a matter of opinion. I get the sense that mutating the flight in other classes is not really stairway's intent (which was my original comment).

Reply:
I don't get what the difference is to stairway when a step is added to a flight. The caller controls what's done so the callee is unaware that stairway is involved at all. From stairway's point of view, the same APIs are called with the same information.

My changes result in less code and remove duplicate code, so I don't see how the current code is simpler. Although simple vs complex can be subjective as well.